### ISSUE TITLE
Can't build garglk with boost 1.75

### ISSUE THREAD

**Issue by lgbaldoni** [garglk/garglk#411]
Trying to fix the garglk package for openSUSE Tumbleweed and this is what I see:
```
+ cd garglk-2019.1.1
+ BUILD=DEBUG
+ CFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -std=gnu++98'
+ CCFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -std=gnu++98'
+ CXXFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -std=gnu++98'
+ OPTFLAGS='-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -std=gnu++98'
+ _BINDIR=/usr/bin
+ _LIBDIR=/usr/lib64
+ _APPDIR=/usr/lib64/garglk
+ jam -d3
warning: mismatched versions of B2 engine and core
warning: B2 engine (jam) is 4.3.0
warning: B2 core (at /usr/share/boost-build/src) is 4.2-git
Jamfile:6: in modules.load
ERROR: rule "SubDir" unknown in module "Jamfile</home/abuild/rpmbuild/BUILD/garglk-2019.1.1>".
/usr/share/boost-build/src/build/project.jam:372: in load-jamfile
/usr/share/boost-build/src/build/project.jam:64: in load
/usr/share/boost-build/src/build/project.jam:142: in project.find
/usr/share/boost-build/src/build-system.jam:618: in load
/usr/share/boost-build/src/kernel/modules.jam:295: in import
/usr/share/boost-build/src/kernel/bootstrap.jam:139: in boost-build
/usr/share/boost-build/boost-build.jam:8: in module scope
```

Am I doing something wrong?

**Comment by cspiegel** [comment]
This looks due to Boost having its own version of jam that isn't compatible with what Gargoyle expects.  The original [Perforce Jam](https://swarm.workshop.perforce.com/files/guest/perforce_software/jam/jam-2.6.1.tar?v=1) works.  The FreeType project claims to have an [improved jam](https://freetype.org/jam/) which is compatible with Perforce.

At some point I plan on moving to CMake, so Jam issues will go away then, but for now you have to use an outdated, probably unmaintained version.

**Comment by lgbaldoni** [comment]
I tried Perforce Jam 2.6.1, ftjam 2.52 and ftjam 2.53rc2: all of them fail linking four executables with several instances of `multiple definition of`.

On the other hand, gargoyle master fails to compile due to apparent incompatibility with SDL2_mixer. That's not on jam, cmake fails in the same way (I created a quick and dirty CMakeLists.txt for that purpose).

**Comment by cspiegel** [comment]
The "multiple definition" is fixed in master.  You may be able to get the fixes into 2019.1.1 by cherry-picking e7038f7 and 91c0b74.  If necessary I can tag a new 2019 release with those fixes, but I'd rather get a new feature release out instead, which brings us to your master issues.

What's the error you're getting on master?

**Comment by lgbaldoni** [comment]
Applying those two patches worked for me. Regarding the rest, I stand corrected: I couldn't reproduce the defect with jam.

So I guess our business here is concluded. TYVM.
